### ISSUE TITLE
Incorrect command still completes tutorial step

### ISSUE THREAD

**Issue by kkiningh** [dhrp/docker-tutorial#1]
I was able to complete the first step of the tutorial by typing "dockerr version" even though the command output in the terminal was "dockerr: command not found."

Looking through the source code it looks like this is related to line 519 of steps.coffee
`if input.containsAllOfTheseParts(_q.command_expected)`
since containsAllOfTheseParts does a partial match on its inputs instead of an exact match.

**Comment by dhrp** [comment]
Thanks for your feedback! Indeed I do partial matching, basically to prevent silly things to make it fail. I'll think about improving the strict parsing for some of the command components.

**Comment by gabceb** [comment]
What about just switching to use containsAllOfThese?

The only silly thing that could be special cased is the use of ' vs "

EDIT: Doing this may also be problematic when using the short docker ID on the tutorial. Hmmmm....

**Comment by dhrp** [comment]
> EDIT: Doing this may also be problematic when using the short docker ID on the tutorial. Hmmmm....

Yup. But since people seem to care I'll see if I can improve it. Will need to make changes anyway with the Docker CLI changes coming up.
